lebedigital/calibration/calibrationWorkflow.py

```diff
@@ -0,0 +1,202 @@
+# third party imports
+import os
+from pathlib import Path
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import matplotlib as mpl
+from matplotlib import rc
+
+mpl.rcParams['font.size'] = 16
+rc('text', usetex=True)
+
+# local imports (probeye)
+from probeye.definition.inverse_problem import InverseProblem
+from probeye.definition.likelihood_model import GaussianLikelihoodModel
+from probeye.ontology.knowledge_graph_export import export_knowledge_graph
+from probeye.ontology.knowledge_graph_import import import_parameter_samples
+from probeye.ontology.knowledge_graph_export import export_results_to_knowledge_graph
+from probeye.inference.emcee.solver import EmceeSolver
+
+# local imports (others)
+from lebedigital.Prediction.three_point_bending_example import three_point_bending_example
+from lebedigital.calibration.utils import query_KG, PosteriorPredictive, load_experimental_data
+from lebedigital.calibration.forwardmodel_linear_elastic_cylinder import LinearElasticityCylinder
+
+
+def perform_calibration(path_to_KG : str, calibrated_data_path : str, experiment_name='Wolf 8.2 Probe 1', mode = 'full', KnowledgeGraph = True):
+    # =========================================
+    #       Loading the experiment
+    # =========================================
+    # for concrete E 60 - 85 Gpa/60 - 85 10^9N/m2 /60-85 KN/mm^2
+    #experiment_name = 'Wolf 8.2 Probe 1'
+    # -- if KG is not there, using custom script
+    # path_csv = '../MinimumWorkingExample/emodul/processed_data/Wolf 8.2 Probe 1.csv'
+    # skip_last = 145
+    # skip_init = 330
+    #experiment_name = 'Wolf 8.2 Probe 1'
+    # exp_output = load_experimental_data(experiment_name, skip_init, skip_last, path=path_csv)
+    # -- if KG is present, comment the above
+    #path_to_KG = '../usecases/MinimumWorkingExample/emodul/knowledge_graphs/'
+    if KnowledgeGraph:
+        path = path_to_KG + '/' + experiment_name + '.ttl'
+        exp_output = query_KG(path=path)
+        plt.plot(exp_output['stress'], exp_output['displacement'] / exp_output['height'])  # checking loading data
+    else:
+        path_csv = path_to_KG # if KG not present, path takes path of CSV
+        #path_csv = os.path.dirname(Path(__file__)) + '/Wolf 8.2 Probe 1.csv'
+        skip_last = 145
+        skip_init = 330
+        exp_output = load_experimental_data(experiment_name, skip_init, skip_last, path=path_csv)
+
+    # This generated Latex errors!!! Maybe we require some extra package
+    #plt.show()
+    # ========================================
+    #       Set Numerical values
+    # ========================================
+    # "uninformed" Normal prior for the E modulus
+    loc_E = 100  # kN/mm2
```

Comment:
How did you get that prior? Shouldn't that be part of the interface?

Reply:
I didnt understand your question. Its a uninformed prior, see the sd is very high. Setting the value of mean is a bit irrelevant in this case.

Reply:
That depends on the number of data points and then weight you are actually giving. And an uninformed prior should have a clear phyiscal meaning, in your case the normal distribution results in values of the Youngs modulus that can be negative, which is not physical.

Reply:
In addition, I suggest that this is not hard coded here, but added by the user (potentially in the doit scripts or in the pytests, but not here in the general routines that should work for any Youngs modulus test.